src/neuroglancer/datasource/computed/tensorflow/frontend.ts

```diff
@@ -0,0 +1,278 @@
+      return Promise.resolve();
+    }
+
+    const prediction = tfjs.tidy(() => {
+      const modelInput = tfjs.tensor(inferenceRequest.array).reshape(this.params.inputTensorShape!);
+      const model = this.model_!;
+      return <Tensor>model.predict(modelInput, {});
+    });
+
+    return prediction.data()
+        .then((outputBuffer: TensorflowArray) => {
+          prediction.dispose();
+          const result: InferenceResult = {outputBuffer};
+          inferenceRequest.resolve(result);
+        })
+        .then(() => {
+          return new Promise<void>((resolve) => {
+            setTimeout(() => resolve(), 10);
+          });
+        });
+  }
+}
+
+export class TensorflowComputationProvider implements VolumeComputationFrontendProvider {
+  getComputation(config: any, {}, params: ComputedVolumeDataSourceParameters) {
+    let modelPath = verifyString(config['modelPath']);
+    if (!modelPath.endsWith('/')) {
+      modelPath += '/';
+    }
+
+    let model: import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs').InferenceModel;
+    let stdDev = 1.0;
+    let mean = 0.0;
+    let outputType = VolumeType.SEGMENTATION;
+
+    if (config['stdDev'] !== undefined) {
+      stdDev = verifyFloat(config['stdDev']);
+    }
+    if (config['mean'] !== undefined) {
+      mean = verifyFloat(config['mean']);
+    }
+    if (config['atype'] !== undefined) {
+      const atype = verifyString(config['atype']);
+      switch (atype) {
+        case 'classifier':
+          break;
+        case 'regressor':
+          outputType = VolumeType.IMAGE;
+          break;
+        default:
+          throw new Error(`Unknown algorithm type ${atype}. Must be "classifier" or "regressor"`);
+      }
+    }
+
+    // Load the model, then do a dummy inference run. This allows us to
+    // explicitly discover the output dimensions, and to compile the
+    // model on the gpu.
+    return loadTFjs()
+        .then(() => {
+          return tfjs.loadFrozenModel(
+              modelPath + 'tensorflowjs_model.pb', modelPath + 'weights_manifest.json');
+        })
+        .then(
+            (tfModel: import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs')
+                 .InferenceModel) => {
+              model = tfModel;
+              if (model.inputs.length !== 1) {
+                throw new Error('Only models with exactly one input are supported');
+              }
+              if (model.outputs.length !== 1) {
+                // Todo: support for multiple-output models.
+                throw new Error('Only models with exactly one output are supported');
+              }
+
+              // Create a blank tensor, run prediction, check output size
+              const dummyOutput = tfjs.tidy(() => {
+                const dummyInput:
+                    import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs').Tensor =
```

Comment:
I think you can just import these type definitions normally via regular non-dynamic import, and as long as you don't use them except in type contexts, they won't actually create a run-time dependency.

Reply:
Done. (Sorry I missed those)